src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,8 +1,34 @@
-import {Component} from '@angular/core';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from 'src/app/shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {}
+export class CardComponent {
+    @Input() product: Product | null = null;
+    @Output() readonly selectedForBuy = new EventEmitter<string>();
+
+    onBuyClick(event: Event) {
+        event.stopPropagation();
+
+        this.selectedForBuy.emit(this.product?._id);
+    }
+
+    showIcon(rating: number | undefined, index: number): boolean {
+        let isClassActive = false;
+
+        if (!rating) {
+            return isClassActive;
+        }
+
+        if (rating >= index + 1) {
+            isClassActive = true;
+        } else {
+            isClassActive = false;
+        }
+
+        return isClassActive;
```

Comment:
Если смотреть в целом, я противник конструкций `if/else`) Рекомендую использовать ранний `return`, этот подход является более читаемым и гибким)

В данном случае все должно уместиться в одно выражение, сохравнять булевое значение не видится смысла)
```suggestion
        return rating && rating >= index + 1;
```

Reply:
Спасибо - отличный вариант. Исправил.
Хотя мне пока удобнее читать первый вариант, но нужно привыкать к правильному сразу )